daemon/logger/loggerutils/rotatefilewriter.go

```diff
@@ -33,45 +44,70 @@ func NewRotateFileWriter(logPath string, capacity int64, maxFiles int) (*RotateF
 	}
 
 	return &RotateFileWriter{
-		f:            log,
-		capacity:     capacity,
-		currentSize:  size,
-		maxFiles:     maxFiles,
-		notifyRotate: pubsub.NewPublisher(0, 1),
+		f:             log,
+		buf:           bytes.NewBuffer(nil),
+		capacity:      capacity,
+		currentSize:   size,
+		lastTimestamp: defaultTimestamp,
+		maxFiles:      maxFiles,
+		notifyRotate:  pubsub.NewPublisher(0, 1),
 	}, nil
 }
 
-//WriteLog write log message to File
-func (w *RotateFileWriter) Write(message []byte) (int, error) {
-	w.mu.Lock()
+//WriteLog write log logEntry to File
+func (w *RotateFileWriter) Write(logEntry *jsonlog.JSONLogs) error {
```

Comment:
What I was thinking here is we instantiate `RoteFileWriter` with a `LogEntryEncoder`, which would be a new type like:

```go
type LogEntryEncoder interface {
    Encode(l LogEntry) error
}
```

And the jsonlog driver would provide an encoder that can convert the `LogEntry` into the jsonlogs encoded format.... which is basically everything that the json log driver's `Log()` method does, but encapsulated into a separate type.

And then change the name of this function to `WriteLog` or `WriteLogEntry` so it's not ever confused with an `io.Writer`.